src/internal.rs

```diff
@@ -326,6 +326,34 @@ where
   }
 }
 
+impl<'a, I, E> Parser<I, char, E> for char
+where
+  I: crate::traits::Slice<crate::lib::std::ops::RangeFrom<usize>>
+    + crate::traits::InputIter
+    + crate::traits::InputLength,
+  <I as crate::traits::InputIter>::Item: crate::traits::AsChar,
+  E: crate::error::ParseError<I>,
```

Comment:
@Stargateur 

It looks like `internal` is a self-contained core of the crate, not depending on anything else.

- Is it ok to use stuff from outside `internal`?
- If not, should this `impl` live somewhere else, like next to `char`?

Once we've decided on how this should look, I'll clean this up and get it implemented for the other types discussed in #1408

Reply:
I can't speak for geal but I think it's ok but I would reverse, move code from `character::char` to here and call this implementation in `character::char`. Trait are ok but maybe parser is too much. Anyway, this file should probably be call `parser` anyway.

> If not, should this impl live somewhere else, like next to char?

Since this impl Parser for a foreign type I think the correct place is here close to trait definition.